downgrade

```diff
@@ -243,25 +268,81 @@ declare -a installed
 cli() {
   while [[ -n "$1" ]]; do
     case "$1" in
+      --pacman)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN="$1"
+        else
+          gettext "Missing --pacman argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --pacman-conf)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN_CONF="$1"
+        else
+          gettext "Missing --pacman-conf argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --arch)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ARCH="$1"
+        else
+          gettext "Missing --arch argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --ala-url)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ALA_URL="$1"
+        else
+          gettext "Missing --ala-url argument"
+          echo
+          gettext "$(usage)"
+          exit 1
+        fi
+        ;;
+      --ala)
+        DOWNGRADE_FROM_ALA=1
+        DOWNGRADE_FROM_CACHE=0
+        ;;
+      --cached)
+        DOWNGRADE_FROM_ALA=0
+        DOWNGRADE_FROM_CACHE=1
+        ;;
+      --nosudo)
+        DOWNGRADE_NOSUDO=1
+        ;;
+      -h | --help)
+        gettext "$(usage)"
+        exit 1
+        ;;
       --)
         shift
         pacman_args=("$@")
         break
         ;;
       *)
         terms+=("$1")
-        shift
         ;;
     esac
+    shift
   done
 
   if ((!"${#terms[@]}")); then
-    (
-      gettext 'usage: downgrade <pkg> [pkg...] [-- pacman_option...]'
-      echo
-      gettext 'see downgrade(8) for details.'
-      echo
-    ) >&2
+    gettext "No packages provided for downgrading"
+    echo
+    gettext "$(usage)"
```

Comment:
This is a behavior change, you're now outputting these errors to `stdout`, not `stderr` like before.

Reply:
Yes true. I will modify error messages to go explicitly to stderr. Sorry, am a bit rusty with my bash :smile:

Committed the change.